Document provisioning steps (#1160)
kyma-incubator/compass: docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md

The provisioning process contains the following steps:
| Name                                   | Domain                   | Description                                                                                                                                     | Owner            |
|----------------------------------------|--------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------|------------------|
| Initialisation                         | Provisioning             | Starts the provisioning process and asks the Director for dashboard_url if the provisioning in Gardener is done.                                | @jasiu001        |

> | Resolve_Target_Secret                  | Hyperscaler Account Pool | Resolves the name of gardener secret with Hypescaler account credentials to be used during cluster provisioning.                                | @koala7659       |

Reviewer: ```suggestion
| Resolve_Target_Secret                  | Hyperscaler Account Pool | Provides the name of a Gardener Secret that contains  Hypescaler account credentials used during cluster provisioning.                                | @koala7659       |
```

| AVS_Configuration_Step                 | AvS                      | AVS Step sets up external and internal monitoring of SKR for the SREs via the Availability Service in SCP.                                      | @abbi-guarav     |
| Create_LMS_Tenant                      | LMS                      | Requests a tenant in the LMS system or provide tenant ID if it was created before.                                                              | @piotrmiskiewicz |
| Provision Azure Event Hubs             | Event Hub                | Creates an Azure event-hub namespace which is a managed kakfa cluster for a kyma-runtime.                                                       | @anishj0shi      |
| Overrides_From_Secrets_And_Config_Step | Kyma overrides           | Configures default overrides for Kyma.                                                                                                          | @jasiu001        |
| ServiceManagerOverrides                | Service Manager          | Configures overrides with Service Manager credentials.                                                                                          | @mszostok        |

> | Request_LMS_Certificates               | LMS                      | Checks if LMS tenant is ready and request certificates. The step configures SKR fluentbit. Requires the Create_LMS_Tenant step executed before. The step does not fail Provisioning Operation. | @piotrmiskiewicz |

Reviewer: ```suggestion
| Request_LMS_Certificates               | LMS                      | Checks if the LMS tenant is ready and requests certificates. The step configures SKR Fluent Bit. It requires the Create_LMS_Tenant step to be completed before. The step does not fail the provisioning operation. | @piotrmiskiewicz |
```

Reviewer: Let's discuss.

> | Create_Runtime                         | Provisioning             | Triggers provisioning in the Provisioner.                                                                                                       | @jasiu001        |

Reviewer: ```suggestion
| Create_Runtime                         | Provisioning             | Triggers provisioning of a Runtime in the Runtime Provisioner.                                                                                                       | @jasiu001        |
```

## Deprovisioning

Each deprovisioning step is responsible for a separate part of cleaning Runtime dependencies. To properly deprovision all Runtime dependencies, you need the data used during the Runtime provisioning. You can fetch this data from the **ProvisioningOperation** struct in the [initialisation](https://github.com/kyma-incubator/compass/blob/master/components/kyma-environment-broker/internal/process/deprovisioning/initialisation.go#L46) step.

Any deprovisioning step shouldn't block the entire deprovisioning operation. Use the `RetryOperationWithoutFail` function from the `DeprovisionOperationManager` struct to skip your step in case of retry timeout. Set at most 5min timeout for retries in your step.

The deprovisioning process contains the following steps:
| Name                         | Domain         | Status      | Description                                                                            | Owner     |
|------------------------------|----------------|-------------|----------------------------------------------------------------------------------------|-----------|

> | Deprovision_Initialization   | Deprovisioning | Done        | Initialize DeprovisioningOperation instance with data read from ProvisioningOperation. | @jasiu001 |

Reviewer: ```suggestion
| Deprovision_Initialization   | Deprovisioning | Done        | Initialize the `DeprovisioningOperation` instance with data fetched from the `ProvisioningOperation`. | @jasiu001 |
```

> | Deprovision Azure Event Hubs | Event Hub      | In progress | Deletes an Azure event-hub namespace.                                                  | @montaro  |

Reviewer: ```suggestion
| Deprovision Azure Event Hubs | Event Hub      | In progress | Deletes the Azure Event Hub Namespace.                                                  | @montaro  |
```

> | Remove_Runtime               | Deprovisioning | Done        | Triggers deprovisioning in the Provisioner.                                            | @jasiu001 |

Reviewer: ```suggestion
| Remove_Runtime               | Deprovisioning | Done        | Triggers deprovisioning of a Runtime in the Runtime Provisioner.                                            | @jasiu001 |
```
